internal/process/vms.go

```diff
@@ -60,21 +58,24 @@ func createVirtualMachineSamples(config *config.Config) {
 				checkError(config.Logrus, ms.SetMetric("datacenterLocation", config.Args.DatacenterLocation, metric.ATTRIBUTE))
 			}
 
-			if cluster, ok := dc.Clusters[vmHostParent]; ok {
-				checkError(config.Logrus, ms.SetMetric("clusterName", cluster.Name, metric.ATTRIBUTE))
+			if c := dc.GetCluster(vmHostParent); c != nil {
+				checkError(config.Logrus, ms.SetMetric("clusterName", c.Name, metric.ATTRIBUTE))
 			}
 
 			if config.IsVcenterAPIType {
 				checkError(config.Logrus, ms.SetMetric("datacenterName", datacenterName, metric.ATTRIBUTE))
 			}
 			checkError(config.Logrus, ms.SetMetric("hypervisorHostname", hostConfigName, metric.ATTRIBUTE))
 
-			resourcePoolName := dc.GetResourcePoolName(vmResourcePool)
-			checkError(config.Logrus, ms.SetMetric("resourcePoolName", resourcePoolName, metric.ATTRIBUTE))
-
+			resourcePool := dc.GetResourcePool(vmResourcePool)
+			if resourcePool != nil {
+				checkError(config.Logrus, ms.SetMetric("resourcePoolName", resourcePool.Name, metric.ATTRIBUTE))
+			}
 			datastoreList := ""
 			for _, ds := range vm.Datastore {
-				datastoreList += dc.Datastores[ds].Name + "|"
+				if d := dc.GetDatastore(ds); d != nil {
+					datastoreList += d.Name + "|"
```

Comment:
same here , we will skip some data about the vm because the datastore has no tags for instance